src/neo/SmartContract/ApplicationEngine.Contract.cs

```diff
@@ -20,17 +22,20 @@ partial class ApplicationEngine
         public static readonly InteropDescriptor System_Contract_NativeOnPersist = Register("System.Contract.NativeOnPersist", nameof(NativeOnPersist), 0, CallFlags.WriteStates);
         public static readonly InteropDescriptor System_Contract_NativePostPersist = Register("System.Contract.NativePostPersist", nameof(NativePostPersist), 0, CallFlags.WriteStates);
 
-        protected internal void CallContract(UInt160 contractHash, string method, CallFlags callFlags, bool hasReturnValue, ushort pcount)
+        protected internal void CallContract(UInt160 contractHash, string method, Array args, CallFlags callFlags)
         {
             if (method.StartsWith('_')) throw new ArgumentException($"Invalid Method Name: {method}");
             if ((callFlags & ~CallFlags.All) != 0)
                 throw new ArgumentOutOfRangeException(nameof(callFlags));
-            if (pcount > CurrentContext.EvaluationStack.Count)
-                throw new InvalidOperationException();
-            StackItem[] args = new StackItem[pcount];
-            for (int i = 0; i < pcount; i++)
-                args[i] = Pop();
-            CallContractInternal(contractHash, method, callFlags, hasReturnValue, args);
+
+            ContractState contract = NativeContract.ContractManagement.GetContract(Snapshot, contractHash);
+            if (contract is null) throw new InvalidOperationException($"Called Contract Does Not Exist: {contractHash}");
+            ContractMethodDescriptor md = contract.Manifest.Abi.GetMethod(method);
+            if (md is null) throw new InvalidOperationException($"Method {method} Does Not Exist In Contract {contractHash}");
+            bool hasReturnValue = md.ReturnType != ContractParameterType.Void;
+
+            if (!hasReturnValue) CurrentContext.EvaluationStack.Push(StackItem.Null);
```

Comment:
Why this push? It won't be throw an Exception at this line?

https://github.com/neo-project/neo-vm/blob/defb2371d9a6a032bdea94e0c8b903a4729c9152/src/neo-vm/ExecutionEngine.cs#L373

Reply:
We push it onto the current stack, so it won't throw.

Reply:
We always return a value in dynamic call just like before.